src/chain/shared_state.rs

```diff
@@ -382,6 +393,16 @@ impl SharedState {
     /// Returns the index of the public key in our_history that will be trusted by the target
     /// Authority
     pub fn proving_index(&self, target: &Authority<XorName>) -> u64 {
+        if let Authority::Node(name) = target {
+            if let Some(info) = self.our_members.get(name) {
+                if let MemberState::Relocating { node_knowledge } = info.state {
+                    // Relocating node, or adult nodes need to be special cased:
+                    // Use the proving index we last know for them.
+                    return node_knowledge;
+                }
+            }
+        }
```

Comment:
What about if the target is `Section` or `PrefixSection` that the relocating node is part of? Shouldn't we handle that as well? Say taking the min of the normal probing index and this `node_knowledge` field?

Reply:
No, for SMD only the elders are targets normally. Other section do not have this information.
If we need something for Elders to communicate to all adults, we'll probably need to tweak it (probably best with a different Autority?).